src/App.tsx

```diff
@@ -15,48 +16,136 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Alfabet = 'alphabet',
+  Length = 'length',
+  Indexdesc = 'reverse Default',
+  Default = 'Default',
+}
+
+const sortAlfabet = (a: string, b: string, direction: boolean) => {
+  return direction === true
+    ? a.localeCompare(b)
+    : b.localeCompare(a);
+};
+
+const sortByLenght = (a: string, b: string, direction: boolean) => {
+  if (a.length === b.length) {
+    return sortAlfabet(a, b, direction);
+  }
+
+  return direction === true
+    ? a.length - b.length
+    : b.length - a.length;
+};
+
+function getSortGoods(
+  goods: string[],
+  sortField: SortBy,
+  sortDirection: boolean,
+) {
+  const preparedGoods: string[] = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortBy.Alfabet:
+          return sortAlfabet(good1, good2, sortDirection);
+
+        case SortBy.Length:
+          return sortByLenght(good1, good2, sortDirection);
+
+        case SortBy.Indexdesc:
+          return preparedGoods.indexOf(good2) - preparedGoods.indexOf(good1);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortBy.Default);
+  const [sortDirection, setSortDirection] = useState(true);
```

Comment:
In my opinion, it will be better to use the `isReversed` state for this purpose. The direction can be `asc` or `desc`. But here is a boolean value https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig